pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,143 @@ $title: Use custom JavaScript in AMP pages
+```
+
+Traditional interactive AMP components may also be more suitable for interactions that span large sections of a webpage - as you may not wish to wrap so much of the DOM in an `<amp-script>`. Generally speaking, `amp-list` and `amp-bind` are tightly integrated with the rest of AMP, and it's straightforward to sprinkle these throughout a webpage.
+
+On the other hand, on pages that involve more complex state variables or multiple interactions, `amp-script` likely provides a simpler path. This example is from the [AMP Camp e-commerce demo site](https://camp.samples.amp.dev/product-details?categoryId=women-shirts&productId=79121):
 
 ```html
-<body>  
-  <p>Hello!</p>
-  <div>
-    <amp-script layout="container" src="customjs.js">
-      <span></span>
-    </amp-script>
-  </div>
-</body>
+<amp-selector
+  name="color"
+  layout="container"
+  [selected]="product.selectedColor"
+  on="select:AMP.setState({
+      product: 
+        {
+          selectedSlide: product[event.targetOption].option - 1,
+          selectedColor: event.targetOption,
+          selectedSize: product[event.targetOption].sizes[product.selectedSize] != null ?
+                        product.selectedSize : 
+                        product[event.targetOption].defaultSize,
+          selectedQuantity: 1
+        }
+      })"
+></amp-selector>
 ```
 
-## Event triggers
+This site was created before `amp-script` was released. But this sort of logic might easily be easier to write and debug in JavaScript. In general, for pages with more business logic, `amp-script` may well be less confusing and will allow you to follow better programming practices.
 
-All event triggers are allowed.
+`amp-script` is designed to control just the portion of a page it encloses. Still, [as noted above](#enhance-amp-components), it can also affect the rest of the page by mutating state variables, just like `amp-bind`.
 
-## API restrictions <a name="api-restrictions"></a>
 
- Some synchronous methods are disallowed in `<amp-script>` and replaced with alternatives, such as [`Element.getBoundingClientRect()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)). Because `Element.getBoundingClientRect()` could not be implemented in a Web Worker, an async alternative to it, `getBoundingClientRectAsync()`, is provided. `getBoundingClientRectAsync()` returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) instead of returning the result directly.
+## You can help
 
-View [this chart](https://github.com/ampproject/worker-dom/blob/main/web_compat_table.md) to see WorkerDOM supported APIs.
+At the time this guide is being written, `amp-script` is still an evolving technology. Thus developers are encouraged to get involved - to think of ways to improve it, to [submit issues](https://github.com/ampproject/amphtml/issues), and of course to [suggest and contribute new features](https://github.com/ampproject/amphtml/pulls)!
```

Comment:
I would update this a bit - AMP itself is still evolving. I would maybe rephrase to:
We encourage AMP developers to get involved! Please submit issue or suggest and contribute new features!